source/production/F0.Minesweeper.Logic/Minefield.cs

```diff
@@ -78,14 +111,30 @@ private void GenerateMinefield(Location clickedLocation)
 				for (uint y = 0; y < height; y++)
 				{
 					allLocations.Add(new Location(x, y));
+					minefield.Add(new Cell(new Location(x, y), false, 0, false));
 				}
 			}
 
 			var mineLocations = mineplacer.PlaceMines(allLocations, mineCount, clickedLocation);
 
-			foreach (var mineLocation in mineLocations)
+			foreach (var minefieldCell in minefield)
 			{
-				minefield[mineLocation.X, mineLocation.Y] = true;
+				if (mineLocations.Any(l => l == minefieldCell.Location))
+				{
+					minefieldCell.IsMine = true;
+					continue;
+				}
+
+				var locationsArea = Utilities.GetLocationsAreaAroundLocation(allLocations, minefieldCell.Location, false);
```

Comment:
In this scenario we should exclude the given location in the utility method. Technically speaking it makes no difference since the current cell is no mine and thus can't effect it's own adjacent mine count. But theoretically speaking it just sounds weird that we look at the current cell as well when we want to know the adjacent mines.

Reply:
Yes that's correct. Thanks! 😄